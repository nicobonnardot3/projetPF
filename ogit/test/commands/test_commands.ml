open Ogitlib
open Commands

(* on se place dans le rÃ©pertoire repo/ *)
let repo_root = "../../../../../repo"
let () = 
  Sys.chdir repo_root

(* on prepare le repertoire .ogit *)
let _ = Sys.command "rm -rf .ogit"


(* test de ogit_init *)
let () = Format.printf "@.EXECUTION DE ogit_init@."
let _ = ogit_init ()

let () = Format.printf "@.NOMBRE DE LOGS : %d@." (Array.length (Sys.readdir ".ogit/logs"))
let () = Format.printf "@.EXECUTION DE LA COMMANDE SHELL ls .ogit/objects@."
let _ = Sys.command "ls .ogit/objects"
let first_commit = (In_channel.with_open_text ".ogit/HEAD" In_channel.input_all)


(* test de ogit_commit *)
let () = Format.printf "@.AJOUT D'UN FICHIER VIDE ET EXECUTION de ogit_commit@."
let _ = Sys.command "echo >empty.txt"
let _ = ogit_commit "vide"

let () = Format.printf "@.EXECUTION DE LA COMMANDE SHELL tree@."
let _ = Sys.command "tree"

let () = Format.printf "@.NOMBRE DE LOGS : %d@." (Array.length (Sys.readdir ".ogit/logs"))
let () = Format.printf "@.EXECUTION DE LA COMMANDE SHELL ls .ogit/objects@."
let _ = Sys.command "ls .ogit/objects"

let second_commit = (In_channel.with_open_text ".ogit/HEAD" In_channel.input_all)

(* test de ogit_checkout *)
let () = Format.printf "@.EXECUTION DE ogit_checkout."
let _ = ogit_checkout first_commit

let () = Format.printf "@.EXECUTION DE LA COMMANDE SHELL tree@."
let _ = Sys.command "tree"

let () = Format.printf "@.NOMBRE DE LOGS : %d@." (Array.length (Sys.readdir ".ogit/logs"))
let () = Format.printf "@.EXECUTION DE LA COMMANDE SHELL ls .ogit/objects@."
let _ = Sys.command "ls .ogit/objects"

(* test de ogit_merge sans conflict  *)
let () = Format.printf "@.AJOUT D'UN FICHIER ET EXECUTION de ogit_commit@."
let _ = Sys.command "echo 42 >foo/rep.txt"

let () = Format.printf "@.EXECUTION de ogit_merge@."
let _ = ogit_merge second_commit

let () = Format.printf "@.NOMBRE DE LOGS : %d@." (Array.length (Sys.readdir ".ogit/logs"))
let () = Format.printf "@.NOMBRE HEAD : %d@." (List.length (Logs.get_head ()))
let () = Format.printf "@.NOMBRE PARENTS : %d@." (List.length (Logs.read_commit (List.hd (Logs.get_head ()))).parents)

let () = Format.printf "@.EXECUTION DE LA COMMANDE SHELL tree@."
let _ = Sys.command "tree"

let _ = Sys.command "rm -f empty.txt"
let _ = Sys.command "rm -f foo/rep.txt" 

(* test de ogit_merge avec conflict  *)
let _ = ogit_checkout first_commit
let () = Format.printf "@.CONFLIT SUR empty.txt@."
let _ = Sys.command "echo plein >empty.txt"

let () = Format.printf "@.EXECUTION de ogit_merge@."
let _ = ogit_merge second_commit

let () = Format.printf "@.NOMBRE DE LOGS : %d@." (Array.length (Sys.readdir ".ogit/logs"))
let () = Format.printf "@.NOMBRE HEAD : %d@." (List.length (Logs.get_head ()))
let () = Format.printf "@.NOMBRE PARENTS : %d@." (List.length (Logs.read_commit (List.hd (Logs.get_head ()))).parents)

let () = Format.printf "@.EXECUTION DE LA COMMANDE SHELL tree@."
let _ = Sys.command "tree"

let _ = Sys.command "rm -f empty.txt*"
let _ = Sys.command "rm -f foo/rep.txt*" 
