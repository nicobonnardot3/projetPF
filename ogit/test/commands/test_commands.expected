
EXECUTION DE ogit_init

NOMBRE DE LOGS : 1

EXECUTION DE LA COMMANDE SHELL ls .ogit/objects
1e423d8532170b358868e1af76688c4a
87113b72a9d03cbfb8aea342455d710f
9ef6d96530cb3791f0dfb16d157220b6
a5b0164846456eb4c32115f57cfc3a38
bb6d4c2870158394ea6b4c6ec804d61f
c3ca6ad75c459ecccfdb10b421cbb489
ec7e8e8ce16f3fe945d3c4c0955315d3

AJOUT D'UN FICHIER VIDE ET EXECUTION de ogit_commit

EXECUTION DE LA COMMANDE SHELL tree
.
├── empty.txt
├── foo
│   ├── bar
│   │   └── tata.txt
│   ├── titi.txt
│   └── toto.txt
└── toto.txt

2 directories, 5 files

NOMBRE DE LOGS : 2

EXECUTION DE LA COMMANDE SHELL ls .ogit/objects
1e423d8532170b358868e1af76688c4a
68b329da9893e34099c7d8ad5cb9c940
789d80b218786e139de0786f0e45e333
87113b72a9d03cbfb8aea342455d710f
9ef6d96530cb3791f0dfb16d157220b6
a5b0164846456eb4c32115f57cfc3a38
bb6d4c2870158394ea6b4c6ec804d61f
c3ca6ad75c459ecccfdb10b421cbb489
ec7e8e8ce16f3fe945d3c4c0955315d3

EXECUTION DE ogit_checkout.
EXECUTION DE LA COMMANDE SHELL tree
.
├── foo
│   ├── bar
│   │   └── tata.txt
│   ├── titi.txt
│   └── toto.txt
└── toto.txt

2 directories, 4 files

NOMBRE DE LOGS : 2

EXECUTION DE LA COMMANDE SHELL ls .ogit/objects
1e423d8532170b358868e1af76688c4a
68b329da9893e34099c7d8ad5cb9c940
789d80b218786e139de0786f0e45e333
87113b72a9d03cbfb8aea342455d710f
9ef6d96530cb3791f0dfb16d157220b6
a5b0164846456eb4c32115f57cfc3a38
bb6d4c2870158394ea6b4c6ec804d61f
c3ca6ad75c459ecccfdb10b421cbb489
ec7e8e8ce16f3fe945d3c4c0955315d3

AJOUT D'UN FICHIER ET EXECUTION de ogit_commit

EXECUTION de ogit_merge

NOMBRE DE LOGS : 3

NOMBRE HEAD : 1

NOMBRE PARENTS : 2

EXECUTION DE LA COMMANDE SHELL tree
.
├── empty.txt
├── foo
│   ├── bar
│   │   └── tata.txt
│   ├── rep.txt
│   ├── titi.txt
│   └── toto.txt
└── toto.txt

2 directories, 6 files

CONFLIT SUR empty.txt

EXECUTION de ogit_merge
CONFLICT (object): Merge conflict in ./empty.txt
Automatic merge failed; fix conflicts and then commit the result

NOMBRE DE LOGS : 3

NOMBRE HEAD : 2

NOMBRE PARENTS : 0

EXECUTION DE LA COMMANDE SHELL tree
.
├── empty.txt..cl
├── empty.txt..cr
├── foo
│   ├── bar
│   │   └── tata.txt
│   ├── titi.txt
│   └── toto.txt
└── toto.txt

2 directories, 6 files
